### PR TITLE
frogger game

### PR DESCRIPTION
**Pull request by IrynaY**
_Complete the form below and remove this line._

### Frogger game (js-oop) 

[Demo](https://irynay.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/IrynaY/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/IrynaY/js-oop/js/app.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    this.sprite = 'images/enemy-bug.png';
+    this.x = x
+    this.y = y
+    this.speed = speed
+};
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt
+};
+Enemy.prototype.render = function() {
+    if(this.x >= 515)
```
Comment (OleksiyRudenko) [inline submissions/IrynaY/js-oop/js/app.js]: Your code already contains some constants on the top.  This is definitely good practice. More constants, please. All "magic numbers" from across entire code base are worth defining as constants. And you definitely know that constants may contain not only scalars but also arrays.

As you follow ES5 OO syntax you are not required to employ `const` and `let`, however some ES6 features will make your code more semantic.

File: submissions/IrynaY/js-oop/js/app.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        this.x = 0
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+function setRandomPosition(allowArray){
+    return allowArray[Math.floor(Math.random() * allowArray.length)]
+}
+
+var Player = function(){
```
Comment (OleksiyRudenko) [inline submissions/IrynaY/js-oop/js/app.js]: `Player` and `Enemy` class have many things (i.e. properties) in common. Consider employing some base prototype class. BTW, `setRandomPosition`, which doesn't make much sense beyond game characters perfectly fits the base class.

File: submissions/IrynaY/js-oop/js/app.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        this.y = setRandomPosition([103, 188, 271])
+    }
+}
+Gem.prototype.render = function() {
+    if(this.visible)
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y, 80, 105)        
+};
+
+var Heart = function(){
+    this.x = setRandomPosition([10, 110, 210, 310, 410])
```
Comment (OleksiyRudenko) [inline submissions/IrynaY/js-oop/js/app.js]: Code repetition. Constants will help avoiding it and also save efforts when you or possible contributors decide to change the code.

File: submissions/IrynaY/js-oop/js/app.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    this.y = setRandomPosition([103, 188, 271])
+
+    this.sprite = 'images/Heart.png'
+    this.visible = true
+}
+Heart.prototype.render = function() {
```
Comment (OleksiyRudenko) [inline submissions/IrynaY/js-oop/js/app.js]: Inheritance helps keeping code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Please, benefit from it.

File: submissions/IrynaY/js-oop/js/app.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                if(this.y - CELL_HEIGHT >= 75)
+                    this.y -= CELL_HEIGHT
+            break;
+            case 'down':
+                if(this.y + CELL_HEIGHT <= 407)
+                    this.y += CELL_HEIGHT
+            break;
+        }
+        if(key === 'space' && this.x > 200 && this.x < 300 && this.y > 350 && this.y < CANVAS_HEIGTH){
+            this.changeSprite(player)
```
[the remark this answers is not in the document]
Comment (IrynaY) [reply submissions/IrynaY/js-oop/js/app.js]: Thanks, I understood the problem, it was a bad idea)

**Pull request merged**